core/config/project_settings.cpp

```diff
@@ -283,6 +277,13 @@ String ProjectSettings::globalize_path(const String &p_path) const {
 bool ProjectSettings::_set(const StringName &p_name, const Variant &p_value) {
 	_THREAD_SAFE_METHOD_
 
+	// Early return if value hasn't changed (unless it's being deleted)
+	if (p_value.get_type() != Variant::NIL) {
+		if (props.has(p_name) && props[p_name].variant == p_value) {
```

Comment:
```suggestion
		VariantContainer *var = props.getptr(p_name);
		if (var && var->variant == p_value) {
```
Avoids double lookup.